sql/mdl.h

```diff
@@ -647,15 +648,23 @@ struct MDL_key {
     wait on resource identified by key.
   */
   const PSI_stage_info *get_wait_state_name() const {
-    return &m_namespace_to_wait_state_name[(int)mdl_namespace()];
+    return &m_namespace_to_wait_state_name[(int)mdl_namespace()].stage_info;
+  }
+
+  const char *get_namespace_name() const {
+    return m_namespace_to_wait_state_name[(int)mdl_namespace()].namespace_name;
   }
 
  private:
   uint16 m_length{0};
   uint16 m_db_name_length{0};
   uint16 m_object_name_length{0};
   char m_ptr[MAX_MDLKEY_LENGTH]{0};
-  static PSI_stage_info m_namespace_to_wait_state_name[NAMESPACE_END];
+  typedef struct {
+    PSI_stage_info stage_info;
+    const char *namespace_name;
+  } MDL_stage_info_wite_name;
+  static MDL_stage_info_wite_name m_namespace_to_wait_state_name[NAMESPACE_END];
```

Comment:
try add const here (if too many errors, ignore this comment)
```suggestion
  static const MDL_stage_info_wite_name m_namespace_to_wait_state_name[NAMESPACE_END];
```